pkg/proxy/iptables/proxier.go

```diff
@@ -241,17 +253,60 @@ func newServiceChangeMap() serviceChangeMap {
 	}
 }
 
-func (scm *serviceChangeMap) update(namespacedName *types.NamespacedName, previous, current *api.Service) {
+func (scm *serviceChangeMap) update(namespacedName *types.NamespacedName, previous, current *api.Service) bool {
 	scm.Lock()
 	defer scm.Unlock()
 
 	change, exists := scm.items[*namespacedName]
 	if !exists {
 		change = &serviceChange{}
-		change.previous = previous
+		change.previous = serviceToServiceMap(previous)
 		scm.items[*namespacedName] = change
 	}
-	change.current = current
+	change.current = serviceToServiceMap(current)
+	if reflect.DeepEqual(change.previous, change.current) {
```

Comment:
This is happening once per endpoint/service watch event.
Given what is happening in syncProxyRules function in terms of both allocations and cpu usage (it's better after recent changes, but it's still a lot), I'm not that surprised.
FYI - 2 weeks ago we used to allocated ~38GB of memory during load test on 100-node clusters - this was significantly more than apiserver was allocating.